pkg/skaffold/runner/runner.go

```diff
@@ -219,11 +219,15 @@ func (r *SkaffoldRunner) buildTestDeploy(ctx context.Context, out io.Writer, art
 
 	// Update which images are logged.
 	for _, build := range bRes {
-		r.imageList.Add(build.Tag)
+		r.imageList.Add(build.Result.Tag)
 	}
 
 	// Make sure all artifacts are redeployed. Not only those that were just built.
-	r.builds = build.MergeWithPreviousBuilds(bRes, r.builds)
+	aRes := make([]build.Artifact, len(bRes))
+	for i, b := range bRes {
+		aRes[i] = b.Result
```

Comment:
As discussed, if there is a build error we should return error here to be in par with the current functionality